gateway-server/src/main/java/org/apache/knox/gateway/services/topology/impl/DefaultTopologyService.java

```diff
@@ -153,17 +146,18 @@ private Topology loadTopology(File file) throws IOException, SAXException, Inter
     return topology;
   }
 
+  @Override
+  public Topology parse(final InputStream content) throws IOException, SAXException {
+    return TopologyUtils.parse(content);
+  }
+
   private Topology loadTopologyAttempt(File file) throws IOException, SAXException {
-    Topology topology;
-    Digester digester = digesterLoader.newDigester();
-    TopologyBuilder topologyBuilder = digester.parse(FileUtils.openInputStream(file));
-    if (null == topologyBuilder) {
-      return null;
-    }
-    topology = topologyBuilder.build();
-    topology.setUri(file.toURI());
-    topology.setName(FilenameUtils.removeExtension(file.getName()));
-    topology.setTimestamp(file.lastModified());
+    Topology topology = parse(FileUtils.openInputStream(file));
```

Comment:
I think it would be cleaner to open/close input stream in the parse method. Basically pass a file reference and not an input stream? 

Otherwise I think we need to make sure to close the input stream?

Reply:
That makes sense.

Reply:
The reason for the TopologyService method taking an InputStream (rather than a File) is that there is not a File available in every case. In one case, it is being used to parse a generated String. There could be two signatures, but I think the InputStream signature supports the two existing cases well enough. The streams must be closed though.